### DISCUSSION TITLE
How to use the load and save file buttons in the Combination Setter window?

### DISCUSSION THREAD

**Discussion by gghlopes** [GrandOrgue/grandorgue#2048] [Help]
I would like to know how to use the Next File, Load File, Save File, etc. buttons. I try clicking them, but nothing happens.

<img width="777" alt="Captura de Tela 2024-11-25 às 04 10 26" src="https://github.com/user-attachments/assets/133ddf01-3b0c-4c70-9e33-8aa76f26a276">

**Comment by oleg68** [comment]
These buttons are designed for working with the combination sets stored in files.

First of all, do File->Export Combination for creating a new combination set and for giving a filename for it. You may create as many combination sets as you need.

Then you can navigate across them with the ``Prev file`` and ``Next file`` buttons, load them with the ``Load file`` button, change in any way with the sequencer window and save them with the ``Save file`` button.

**Comment by gghlopes** [comment]
I tried to do this, but it still seems like the buttons don't work. I click but nothing happens.

**Reply by hnb2907** [reply]
Not quite as @oleg68 mentioned, I use the "File / Export combinations" instead of the "Save" button.  
Also check the default path is ok, and you have your saved combination files in there.  Then you should be able to use next/prev/load to recall them.  I seem to remember I had some problems with this a long time ago, probably something with file/directory permissions in linux.


This maybe useful for you, if you have a console with 2 available buttons.  On mine, I have 2 buttons "+" and"-".

For the "+" button, it generates a MIDI note on message, which is captured by "Next file" in GO.  This button also sends MIDI note off, which is captured by "Load file" in GO.
Similarly for "-"button, where MIDI note on message is captured by "Prev file", and note off by "Load file".

These are easily configured by right click on the GO buttons shown in the screenshot.

So my console can automatically select the next/previous combination file in the default combination directory, and loads it; just by pressing the "+" and "-" buttons on the console :)

Cheers,
Chris.

**Reply by oleg68** [reply]
> I tried to do this, but it still seems like the buttons don't work. I click but nothing happens.

Do you see the combinations set filename in the display between the buttons?

**Comment by gghlopes** [comment]
Now it was, there could be a button that imports and export the file. So that it is possible to do it with a midi button

**Reply by oleg68** [reply]
> Now it was, there could be a button that imports and export the file. So that it is possible to do it with a midi button

`Load file` - imports the file, `Save file` - exports the file, that's name is displayed on the sequencer label.

The main problem with a midi button is to enter a filename. Choosing directories is also difficult with midi buttons.

So at the first you should do it with the menu File->Export Combinations and enter the filename.

For choosing a directory you have to do File->Import combinations with the menu. Once having done it, you can navigate over other files **in the same directory** and import/export them only with midi buttons.

If all your combinations are there in the default directory (File->Settings->Paths->Combinations), you can export and import them only with midi without interventing with the menu.

**Reply by gghlopes** [reply]
In fact, I wouldn't even use a midi controller, I would just like to have the option of adding a button to import and another to save in the sample window that I'm creating with GoOdf, that way in the same window of the records I would have the possibility to save and import a combination quickly.

**Reply by oleg68** [reply]
@guilhermelopesdev It is already possible to add any setter buttons to an ODF-created panel. But how are you going to provide the combination set filename for these button?

One possible option is to add all four buttons and the filename display to the ODF panel.

**Reply by gghlopes** [reply]
I would like to have a button in the application window that, when pressed, opens the file explorer to choose where to save and also specify the file name. The same applies to import combinations. Is there a way to do this?

**Reply by oleg68** [reply]
No, there is not. Use the File menu instead.

Most of buttons have been desired for midi input so they do not open the file explorer. The setter buttons have their own "single directory" mini-explorer, that can be midified easy .

**Reply by rousseldenis** [reply]
@guilhermelopesdev Has this solved your issue ?
